### PR TITLE
Test and fix edge cases in Quotes

### LINKED ISSUE #127: Avoid `identifier expected but $XMLSTART$< found` error from Cross-Quotes on Scala 2
It seems that if we do:

```scala
val expr: Expr[SomeType] = ...

Expr.quote {
  Expr.splice(expr)
}
```

if there is no `Type[SomeType]` available in the scope, on Scala 2 we would get an error about badly parsed $XML$ - probably there is an unhandled case, where missing implicit generate invalid quasiquote interpolation (wrong `$`) that is mistakenly interpreted as bad XML literal.

### LINKED ISSUE #133: Chaining on Expr.splice inside Expr.quote sometimes is not handled by Cross-Quotes on Scala 2
This piece of code in some demo implementation:

```scala
Expr.quote {
  val _ = Expr.splice(ctx.sb)
    .append(Expr.splice(name))
    .append("(\n")
    .append(Expr.splice(Expr(fieldName)))
    .append(" = ")
  Expr.splice(fieldResult)
}
```

generates this code:

```scala
FastShowPrettyMacrosImpl.this .Expr.splice[StringBuilder](renderPreviousFields).append(",\n").append(FastShowPrettyMacrosImpl.this .Expr.splice[String](FastShowPrettyMacrosImpl.this .Expr.apply[String](fieldName)(FastShowPrettyMacrosImpl.this .ExprCodec.StringExprCodec))).append(" = ")
```

which result in this error:

```
FastShowPrettyMacrosImpl is not an enclosing class
```

If `renderPreviousFields` was interpolated into the `Tree`, it would work as expected.

### PR DESCRIPTION
**Pull request by MateuszKubuszok**
- [x] refactor Expr tests
- [x] fix #127
- [x] fix #133 
- [x] test Exprs for using implicits where necessary:
  - [x] from type bound
  - [x] from implicit param
  - [x] from import
  - [x] from declared def

---

> [!NOTE]
> Improves macro handling for Expr.quote/splice (implicit Type resolution, correct printing/escaping) and refactors/adds crossquotes tests; also enhances error messages and import alias printing.
> 
> - **Macros (Scala 2)**:
>   - **Expr.quote/splice**: infer or synthesize implicit `Type[A]`, pass `WeakTypeTag` explicitly, and replace splices using pretty-printed code with `$`-safe escaping; improved logging.
>   - **Type.of**: add imports to sanitize printed types; better handling of imported `Underlying` cases during type injection; richer error location in `reportIssue`.
>   - **Quasiquote conversion**: adjust pipeline order and log parsed source; print `weakTypeTag` stubs via `showCodePretty`.
> - **Pretty Printer**:
>   - Handle imported alias in `Select(Ident(_), name)` so `import ee.{value as expr}` prints `expr` (not `ee.expr`).
> - **Tests/Fixtures**:
>   - Refactor from `typed` to `crossquotes`; add shared `CrossExprsFixturesImpl` and new fixtures for Scala 2/3.
>   - Add `CrossExprsSpec` covering unsanitized blocks, multiple splices, generics via type-bound/implicit/import/declared val, nested quotes/splices, and chaining edge case.
>   - Update `CrossTypesFixtures`/`CrossTypesSpec` package and names accordingly.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 074637b636eeb010bba86a866a94b22c9f3c16db. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: hearth-cross-quotes/src/main/scala-2/hearth/cq/CrossQuotesMacros.scala [hunk 1]
```diff
@@ -2942,15 +2946,39 @@ final class CrossQuotesMacros(val c: blackbox.Context) extends ShowCodePrettySca
...
+          if (loggingEnabled) {
+            println(s"""No implicit Type[$A] found in the scope, creating one with Type.of[$A]""")
+          }
+          val resolved = typeOfImpl[A]
+          if (loggingEnabled) {
+            println(s"""Created Type[$A] with: ${showCodePretty(resolved, SyntaxHighlight.ANSI)}""")
+          }
+          resolved
+      }
+    val quasiquote = convert(ctx)(expr.tree)
```
[the remark this answers is not in the document]
Comment (MateuszKubuszok) [reply hearth-cross-quotes/src/main/scala-2/hearth/cq/CrossQuotesMacros.scala]: BS: `silent = false` turns missing implicit into an exception, which I confirmed looking at the println debug logs and passing tests.
